CIP-0159/README.md

```diff
@@ -0,0 +1,587 @@
+Cardano can unlock new use cases **without sacrificing local determinism**. And by still requiring
+UTxO inputs in transactions, these accounts avoid many of the pitfalls from Ethereum-style accounts.
+
+## Motivation: why is this CIP necessary?
+
+Consider the following use cases:
+
+### Wallet Revenue
+
+Imagine a Cardano wallet that wants to charge a fee in a transaction it generates for a user. Right
+now, the `minUTxOValue` network parameter (now called `utxoCostPerByte`) disallows UTxOs with less
+than ~1 ADA. So if the wallet just had the user send the money to them directly, **the smallest fee
+they could charge is ~1 ADA**. When the average transaction fee is only about [0.39
+ADA](https://cardanoscan.io/analytics/dailyTxCountAndFees), it is hard to justify paying an extra 1
+ADA on top.
+
+*What if the wallet wanted to charge only 0.1 ADA? How could it get this 0.1 ADA from the user?*
+
+The current method is to have the user deposit 1 ADA into a UTxO for the wallet provider. Then,
+whenever the user needs to pay 0.1 ADA, this *deposit UTxO* would be spent to accumulate the 0.1 ADA.
+This UTxO is effectively a "fee bucket" for that user.
+
+While this method works, it has a few downsides:
+
+1. It creates a bad UX since features are now gated behind different deposits that need proper
+   management (with smart contracts for trustlessness).
+2. What if the user wants to close out their deposit UTxO? If the deposit UTxO has less than 2 ADA
+   (e.g., 1.5 ADA), the amount above 1 ADA (e.g., 0.5 ADA) needs to be sent to the wallet provider
+which isn't possible due to the `minUTxOValue`! This complicates the reclamation process for user
+deposits.
+3. The wallet provider needs to manage one deposit UTxO per user which doesn't scale: withdrawing
+   the accumulated fees requires spending one UTxO per user! (Having fewer UTxOs than users will
+cause UTxO contention.)
+
+**All of these problems go away if the wallet provider can just have the user deposit 0.1 ADA into
+the designated account address!** By being able to charge much smaller fees, Cardano wallets will
+have a significantly easier time finding product-market fit.
+
+> [!IMPORTANT]
+> In order for direct deposits to actually be useful, *partial withdrawals are a fundamental
+> requirement*. To see why, imagine a wallet charges a 0.1 ADA fee per transaction and these
+> transactions are submitted every 30 seconds. Currently, the ledger rules require the wallet to
+> withdraw the full balance in the account address. Let's say when the wallet submits the withdrawal
+> transaction, the balance is 100 ADA. While this transaction is sitting in the mempool, another
+> user directly deposits another 0.1 ADA into the wallet's account address. Now the wallets
+> transaction in the mempool is *not* withdrawing the full amount and will fail phase 1 validation!
+> Without partial withdrawals, whether the wallet will actually be able to access the assets in the
+> account will depend on luck!
```

Comment:
We need to mention here that partial withdarwals will become available, but only when PlutusV1-V3 scripts are not present in the transaction, since we cannot change the behavior of older scripts.